go/src/lib/lomcommon/helper.go

```diff
@@ -64,41 +78,51 @@ func SetLogLevel(lvl syslog.Priority) {
  * Return:
  *  None
  */
-func LogMessage(lvl syslog.Priority, s string, a ...interface{})  {
-    prefix := ""
-    if _, fl, ln, ok := runtime.Caller(2); ok {
-        l := strings.Split(fl, "/")
-        c := len(l)
-        if c > 2 {
-            c -= 1
-        }
-        prefix = fmt.Sprintf("%s:%d:", strings.Join(l[c-1:], "/"), ln)
-    }
+func LogMessageWithSkip(skip int, lvl syslog.Priority, s string, a ...interface{}) string {
     ct_lvl := GetLogLevel()
+    m := fmt.Sprintf(getPrefix(skip+2)+s, a...)
     if lvl <= ct_lvl {
-        FmtFprintf(writers[lvl], prefix+s, a...)
+        FmtFprintf(writers[lvl], m)
         if ct_lvl >= syslog.LOG_DEBUG {
             /* Debug messages gets printed out to STDOUT */
-            fmt.Printf(prefix+s, a...)
+            fmt.Printf(m)
             fmt.Println("")
         }
     }
+    return m
+}
+
+
+func LogMessage(lvl syslog.Priority, s string, a ...interface{}) string {
+    return LogMessageWithSkip(2, lvl, s, a...)
 }
 
 
 /* Log this message for panic level and exit */
-func LogPanic(s string, a ...interface{})  {
-    LogMessage(syslog.LOG_CRIT, s, a...)
+func LogPanic(s string, a ...interface{}) string {
+    m := LogMessage(syslog.LOG_CRIT, s, a...)
     LogMessage(syslog.LOG_CRIT, "LoM exiting ...")
```

Comment:
You can combine line 103+104  ..m := LogMessage(syslog.LOG_CRIT, s + "LOM Exiting ...", a...)

Reply:
done